packages/terra-form-select/src/Select.jsx

```diff
@@ -197,11 +203,20 @@ class Select extends React.Component {
   render() {
     const { intl } = this.context;
     const {
-      children, defaultValue, onChange, placeholder, value, ...otherProps
+      allowClear, children, defaultValue, onChange, placeholder, value, ...otherProps
     } = this.props;
 
     const defaultPlaceholder = intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
     const selectPlaceholder = placeholder === undefined ? defaultPlaceholder : placeholder;
+    let clearOptionDisplay;
+
+    if (allowClear) {
+      if (selectPlaceholder.length === 0) {
```

Comment:
If they give us an empty string, do we really want to override that?

Reply:
We use the placeholder as the display for the clear option. If they provide an empty placeholder, we don't want the display for the clear option to also be empty. We are not overriding their placeholder, we are just falling back to our default placeholder (" -Select- ") as the clear option display if the placeholder they provide is really empty.